src/command_modules/azure-cli-webapp/azure/cli/command_modules/webapp/custom.py

```diff
@@ -62,11 +75,37 @@ def get_app_settings(resource_group, name, slot=None):
     result = _generic_site_operation(resource_group, name, 'list_site_app_settings', slot)
     return result.properties
 
-def update_site_configs(resource_group, name, name_value_pairs, slot=None):
+def update_site_configs(resource_group, name,
+                        php_version=None, python_version=None,
+                        net_framework_version=None,
+                        java_version=None, java_container=None, java_container_version=None,
+                        remote_debugging_enabled=None, web_sockets_enabled=None,
+                        always_on=None, auto_heal_enabled=None,
+                        use_32bit_worker_process=None,
+                        slot=None):
     configs = get_site_configs(resource_group, name, slot)
-    for name_value in name_value_pairs:
-        name, value = name_value.split('=')
-        setattr(configs.properties, name, value)
+    if php_version is not None:
+        configs.php_version = php_version
+    if python_version is not None:
+        configs.python_version = python_version
+    if net_framework_version is not None:
+        configs.net_framework_version = net_framework_version
+    if java_version is not None:
+        configs.java_version = java_version
+    if java_container is not None:
+        configs.java_container = java_container
+    if java_container_version is not None:
+        configs.java_container_version = java_container_version
+    if remote_debugging_enabled is not None:
+        configs.remote_debugging_enabled = remote_debugging_enabled
+    if web_sockets_enabled is not None:
+        configs.web_sockets_enabled = web_sockets_enabled
+    if always_on is not None:
+        configs.always_on = always_on
+    if auto_heal_enabled is not None:
+        configs.auto_heal_enabled = auto_heal_enabled
+    if use_32bit_worker_process is not None:
+        configs.use32_bit_worker_process = use_32bit_worker_process
```

Comment:
Could this not be reduced using some reflection type methods?

for key in signature.keys(): #warning pseudocode!
  setattr(configs, key, None)

Since the values all map 1-to-1 and are on the same object.